### ISSUE TITLE
Alerting: math expression dont work as expected when some variables have no values

### ISSUE THREAD

**Issue by romkimchi1002** [grafana/grafana#74636]
### What happened?

![image](https://github.com/grafana/grafana/assets/116630343/eca7e278-533e-4db4-9793-2026e1482b51)

trying to use math expression on 2 vars when one of them has no value

### What did you expect to happen?

alert on site=dummy

### Did this work before?

no

### How do we reproduce it?

1. have 2 reducers, one empty and one that have data
2. create math expression that contains both (doesnt matter how can be || && or anything else )

### Is the bug inside a dashboard panel?

no

### Environment (with versions)?

Grafana: tried on both 9.5.2 and 10.1
OS: ubuntu
Browser: chrome

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

### Datasource(s)?

not specific

**Comment by mellieA** [comment]
@romkimchi1002 this looks like you could be describing the same problem explained here: 

https://github.com/grafana/grafana/issues/57961

**Comment by gillesdemey** [comment]
I agree that this behaviour is confusing and not what you would expect. When any dependent node of an expression is `no data` the evaluation of that node is skipped, so the math expression in this case is never evaluated.

Changing the behaviour of the expressions with `no data` results would be a feature request we can look in to.

**Comment by romkimchi1002** [comment]
thanks, is there any workaround for that problem?

**Comment by romkimchi1002** [comment]
> I agree that this behaviour is confusing and not what you would expect. When any dependent node of an expression is `no data` the evaluation of that node is skipped, so the math expression in this case is never evaluated.
> 
> Changing the behaviour of the expressions with `no data` results would be a feature request we can look in to.

Hi!
is there any plan to add this feature?
